src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Windows.cs

```diff
@@ -42,7 +42,7 @@ public unsafe abstract partial class FileSystemEnumerator<TResult> : CriticalFin
         public FileSystemEnumerator(string directory, EnumerationOptions options = null)
         {
             _originalRootDirectory = directory ?? throw new ArgumentNullException(nameof(directory));
-            _rootDirectory = PathHelpers.TrimEndingDirectorySeparator(Path.GetFullPath(directory));
+            _rootDirectory = PathInternal.TrimEndingDirectorySeparator(Path.GetFullPath(directory));
```

Comment:
**Unrelated question:** Does this entire class need to be marked as unsafe? It looks like the methods that need to be unsafe are already marked as such. And if we need the class to be unsafe, can we remove the redundant unsafe on some of the methods?

Reply:
I'll make a point of auditing the unsafe keywords next time I hit this class.